### PR TITLE
[#860] Adding Spurious Correlation feature

### PR DESCRIPTION
**Pull request by allincowell**
## Summary

> 🎯 **Purpose**: Adding Spurious Correlation feature for Image datasets.
>
> 📜 **Example Usage**: Finds correlation score between one of the image properties like dark score, blurry score, information score, size, aspect_ratio, etc. and the class labels using certain metrics like baseline accuracy and held-out accuracy by fitting a univariate model.

**[ ✏️ Write your summary here. ]**

1. Added `spurious_correlation.py` module in `cleanlab/datalab/internal` location.
2. Added a private instance method `_spurious_correlation` in `Datalab` class that uses an instance of `SpuriousCorrelations` class.

## Links to Relevant Issues or Conversations

> Issue Link: https://github.com/cleanlab/cleanlab/issues/860
> Early PR attempted: https://github.com/cleanlab/cleanlab/pull/872

### CODE REVIEW

File: cleanlab/datalab/datalab.py [hunk 1]
```diff
@@ -635,3 +636,64 @@ def load(path: str, data: Optional[Dataset] = None) -> "Datalab":
         load_message = f"Datalab loaded from folder: {path}"
         print(load_message)
         return datalab
+
+    def _spurious_correlation(
+        self, properties_of_interest: Optional[List[str]] = None
```
Comment (elisno) [inline cleanlab/datalab/datalab.py:641]: For now, let's omit this argument in `Datalab._spurious_correlation()`.

Remember to remove the parameter in the docstring as well.

File: cleanlab/datalab/datalab.py [hunk 2]
```diff
@@ -635,3 +636,64 @@ def load(path: str, data: Optional[Dataset] = None) -> "Datalab":
...
+        odd_size_score          0.900000
+        low_information_score   0.742617
+        dark_score              0.855250
+        grayscale_score         0.900000
+        light_score             0.900000
+        blurry_score            0.747367
+        odd_aspect_ratio_score  0.900000
+        """
+        try:
+            issues = self.get_issues()
```
Comment (elisno) [inline cleanlab/datalab/datalab.py:682]: Please add a validation step that ensures that the `issues` dataframe has all the relevant (image-specific) scores.
If it doesn't an error with a helpful message should be raised.

File: cleanlab/datalab/datalab.py [hunk 3]
```diff
@@ -635,3 +636,64 @@ def load(path: str, data: Optional[Dataset] = None) -> "Datalab":
...
+        >>> # Assuming `find_issues()` has already been called
+        >>> datalab._spurious_correlation()
+        property                score
+        odd_size_score          0.900000
+        low_information_score   0.742617
+        dark_score              0.855250
+        grayscale_score         0.900000
+        light_score             0.900000
+        blurry_score            0.747367
+        odd_aspect_ratio_score  0.900000
```
Comment (elisno) [inline cleanlab/datalab/datalab.py:679]: Let's simplify this for now.


```suggestion
        Assess potential spurious correlations in issue severity scores.

        This method calculates scores indicating the likelihood of spurious correlations 
        for various issue severity scores in the dataset, as estimated by the `find_issues()` method.
        Currently, it focuses on severity scores related to image attributes.
        If `find_issues()` has not been called, it raises a ValueError.


        Returns
        -------
        correlations_df :
            A DataFrame containing the spurious correlation scores for each property, excluding 
            'class_imbalance_score'. The DataFrame includes:
            - 'property': The name of the property.
            - 'score': The spurious correlation score (between 0 and 1) for the property, where a low score indicates a higher likelihood of spurious correlation, and a high score indicates a lower likelihood.

        Raises
        ------
        ValueError
            If the issues have not been identified (i.e., `find_issues()` has not been called).
            
        Notes
        -----
        This method currently focuses on image-related severity scores, with potential for future 
        expansions.
```

File: cleanlab/datalab/datalab.py (hunk 2 above)
Comment (allincowell) [reply cleanlab/datalab/datalab.py:671]: Added a validation step [here](https://github.com/cleanlab/cleanlab/pull/1140/files#diff-e126cf653842496e91462249d07fa417cca997b74bc9924d4c80d9c715c11ea0R687) to cjeck all vision/image issues are present in the correlations dataframe

Comment (elisno) [reply cleanlab/datalab/datalab.py:671]: To be clear, the `issues` dataframe should be validated, not the `correlations_df`.

**Review by elisno — APPROVED** [review]

**Pull request merged**